src/main.rs

```diff
@@ -92,13 +104,43 @@ fn run() -> Result<(), Box<std::error::Error>> {
     {
```

Comment:
The contents of this block could be put into a seperate function which is appropriately named.